dot/core/service.go

```diff
@@ -454,26 +454,27 @@ func (s *Service) DecodeSessionKeys(enc []byte) ([]byte, error) {
 }
 
 // GetRuntimeVersion gets the current RuntimeVersion
-func (s *Service) GetRuntimeVersion(bhash *common.Hash) (runtime.Version, error) {
+func (s *Service) GetRuntimeVersion(bhash *common.Hash) (
+	version runtime.Version, err error) {
```

Comment:
did you consider returning a `*runtime.Version`.  I feel like the zero case of nil is more applicable here.

Reply:
I think we should keep it dumb without pointer whenever possible (and for performance as well).
If an error happens, the caller should discard the version returned anyway and handle the error.